src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -1726,6 +1726,46 @@ private async Task<CompletionItem[]> ApplyCompletionExtensions(ConnectionInfo co
 
         #region Diagnostic Provider methods
 
+        /// <summary>
+        /// Checks for non T-SQL syntax within the Parse Result, and 
+        /// sends notification if non T-SQL syntax is detected
+        /// Public for testing purposes
+        /// </summary>
+        /// <param name="uri"></param>
+        /// <param name="parseResult"></param>
+        public async Task<bool> CheckForNonTSqlLanguage(string uri, ParseResult parseResult)
+        {
+            if (parseResult.Errors.Count() >= TSqlDetectionConstants.SqlFileErrorLimit)
+            {
+                await ServiceHostInstance.SendEvent(
+                                   NonTSqlNotification.Type,
+                                   new NonTSqlParams
+                                   {
+                                       OwnerUri = uri,
+                                       ContainsNonTSqlKeywords = false
+                                   });
+
+                return true;
+            }
+            var significantTokenTexts = parseResult.Script.Tokens
+             .Where(token => token.IsSignificant)
+             .Select(token => token.Text.ToUpperInvariant())
+             .ToList();
```

Comment:
I think this is an important path in which to pay attention to performance because we want this to be repsonsive.  Linq is a wonderful tool, but ends up creating new lists each time you chain a call.  Since you only care about finding a single keyword, you can tighten the performance here a couple ways:

* Initialize the HashSet with a case-insensitive comparer to avoid needing to call `ToUpperInvariant()` on every token
* convert the chained Linq calls to a for-loop that checks `IsSignificant` and then immediately if the HashSet contains that keyword.